### PR TITLE
CORE: Cure cast time reductions

### PR DESCRIPTION
**Pull request by Jupiter065**
White mage merit cure cast reduction
item bonus cure cast reduction

### CODE REVIEW

File: src/map/ai/states/magic_state.cpp [hunk 1]
```diff
@@ -253,6 +253,16 @@ uint32 CMagicState::CalculateCastTime(CSpell* PSpell)
...
     }
+    if (PSpell->isCure())
+    {
+        int32 cureCast = m_PEntity->getMod(MOD_CURE_CAST_TIME);
+        if (m_PEntity->objtype == TYPE_PC)
+        {
+            cureCast += ((CCharEntity*)m_PEntity)->PMeritPoints->GetMeritValue(MERIT_CURE_CAST_TIME, (CCharEntity*)m_PEntity);
+        }
+        cureCast = dsp_cap(cureCast, 0, 100);
+        cast = cast * ((100.0f - cureCast) / 100.0f);
```
Comment (teschnei) [inline src/map/ai/states/magic_state.cpp]: you're going to have to verify that cure cast is multiplicative with fast cast and not additive

Comment (Jupiter065) [inline src/map/ai/states/magic_state.cpp]: Ok, what would the cap be then? 80% like in that thread?

**Comment by Jupiter065** [comment]
Ok, I've made it additive with fast cast and capped it at 80%.

**Comment by teschnei** [comment]
yep, 80% is right

**Pull request merged**